pkg/build/docker_go.go

```diff
@@ -407,28 +549,36 @@ COPY . /
 
 RUN cd ${PLAN_DIR} \
     && go env -w GOPROXY="${GO_PROXY}" \
-    && GOOS=linux GOARCH=amd64 go build -o /testplan ${BUILD_TAGS} ${TESTPLAN_EXEC_PKG}
+    && GOOS=linux GOARCH=amd64 go build -o ${PLAN_DIR}/testplan.bin ${BUILD_TAGS} ${TESTPLAN_EXEC_PKG}
 
 {{.DockerfileExtensions.PostBuild}}
 
 # Store module dependencies
 RUN cd ${PLAN_DIR} \
   && go list -m all > /testground_dep_list
 
-{{ if not .SkipRuntimeImage }}
 #:::
-#::: RUNTIME CONTAINER
+#::: (OPTIONAL) RUNTIME CONTAINER
 #:::
 
-FROM ${RUNTIME_IMAGE} AS binary
+{{ if not .SkipRuntimeImage }}
+
+FROM ${RUNTIME_IMAGE} AS runtime
+
+# PLAN_DIR is the location containing the plan source inside the build container.
+ENV PLAN_DIR /plan
 
 {{.DockerfileExtensions.PreRuntimeCopy}}
 
 COPY --from=builder /testground_dep_list /
-COPY --from=builder /testplan /
+COPY --from=builder ${PLAN_DIR}/testplan.bin /testplan
 
 {{.DockerfileExtensions.PostRuntimeCopy}}
 
+{{ else }}
+
+FROM builder AS runtime
```

Comment:
Isn't this redundant? If the previous stage is the `builder`, why do we need this line?

Reply:
To find the marker to extract the image ID of the last step of the builder image. Docker makes us parse stdout... (line 470).

Reply:
Actually I see it is used as a pointer, if that's the case, we probably want a comment on it.